### PR TITLE
Start some real documentation

### PR DESCRIPTION
**Pull request by c42f**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,8 +1,61 @@
...
+## Progress bars
+
+`TerminalLogger` displays progress logging as a set of progress bars which are
+cleanly separated from the rest of the program output at the bottom of the
+terminal. Try an example like the following
+
+```julia
+global_logger(TerminalLogger(stderr, right_justify=120))
+for i=1:100
+    @info "Some progress" progress=i/100
```
Comment (tkf) [inline docs/src/index.md]: Strictly speaking, it's more "correct" to use `progress=(i-1)/100`, isn't it? Or maybe move this after `sleep(0.01)`?

This sometimes leads to a bit annoying situation where the progress bar says 100% but the last job is not quite done yet.

File: docs/src/index.md [hunk 2]
```diff
@@ -1,8 +1,61 @@
...
+To use `TerminalLogger` in all your REPL sessions by default, you may add a
+snippet such as the following to your startup file (in `.julia/config/startup.jl`)
+
+```julia
+atreplinit() do repl
+    try
+        @eval begin
+            import Logging: global_logger
+            import TerminalLoggers: TerminalLogger
+            global_logger(TerminalLogger(stderr))
```
Comment (tkf) [inline docs/src/index.md]: Why not

```suggestion
            global_logger(TerminalLogger())
```

?

File: docs/src/index.md [hunk 3]
```diff
@@ -1,8 +1,61 @@
...
+
+
+## Progress bars
+
+`TerminalLogger` displays progress logging as a set of progress bars which are
+cleanly separated from the rest of the program output at the bottom of the
+terminal. Try an example like the following
+
+```julia
+global_logger(TerminalLogger(stderr, right_justify=120))
```
Comment (tkf) [inline docs/src/index.md]: ditto

```suggestion
global_logger(TerminalLogger(right_justify=120))
```

File: src/TerminalLogger.jl [hunk 4]
```diff
@@ -2,8 +2,9 @@
     TerminalLogger(stream=stderr, min_level=$ProgressLevel; meta_formatter=default_metafmt,
                    show_limited=true, right_justify=0)
 
-Logger with formatting optimized for readability in a text console, for example
-interactive work with the Julia REPL.
+Logger with formatting optimized for interactive readability in a text console
+(for example, the Julia REPL). This is an enhanced version of the terminal
+logger `Logging.ConsoleLogger` which comes installed in the REPL by default.
```
Comment (tkf) [inline src/TerminalLogger.jl]: It looks like `ConsoleLogger` is used outside the `REPL` stdlib too?

```console
$ julia --startup-file=no -e "using Logging; @show typeof(current_logger())"
typeof(current_logger()) = ConsoleLogger
```

Comment (c42f) [reply src/TerminalLogger.jl]: True, it's the default which is applied in the stdlib `Logging.__init__()` (which in turn replaces a `CoreLogging.SimpleLogger` instance).

File: docs/src/index.md (hunk 1 above)
Comment (c42f) [reply docs/src/index.md]: Definitely. Will fix.

I feel like we should modify this shortly to use your new progress type anyway :-)

File: src/TerminalLogger.jl (hunk 4 above)
Comment (c42f) [reply src/TerminalLogger.jl]: Tweaked to fix this.

File: docs/src/index.md [hunk 5]
```diff
@@ -1,8 +1,60 @@
...
+```
+
+To use `TerminalLogger` in all your REPL sessions by default, you may add a
+snippet such as the following to your startup file (in `.julia/config/startup.jl`)
+
+```julia
+atreplinit() do repl
+    try
+        @eval begin
+            import Logging: global_logger
```
Comment (tkf) [inline docs/src/index.md]: Let's avoid accidentally overloading `global_logger`.

```suggestion
            using Logging: global_logger
```

File: docs/src/index.md [hunk 6]
```diff
@@ -1,8 +1,60 @@
...
+            import TerminalLoggers: TerminalLogger
```
Comment (tkf) [inline docs/src/index.md]:
```suggestion
            using TerminalLoggers: TerminalLogger
```

**Comment by c42f** [comment]
Well these docs could certainly be better but this package doesn't do a whole lot yet :-)

For now I'll merge this once CI passes and I suggest we release version 0.1.

**Pull request merged**